### PR TITLE
Add text classification docs

### PR DESCRIPTION
**Pull request by VahidooX**

### CODE REVIEW

File: docs/source/asr/results.rst [hunk 1]
```diff
@@ -63,6 +63,19 @@ If you would like to programatically list the models available for a particular
...
+Transcribing/Inference
+^^^^^^^^^^^^^^^^^^^^^^
+
+You may perform inference and transcribe a sample of speech after loading the model by using its 'transcribe()' method:
+
+.. code-block:: python
+
+    model.transcribe(paths2audio_files=[list of audio files], batch_size=BATCH_SIZE, logprobs=False)
+
+Setting argument 'logprobs' to True would return the log probabilities instead of transcriptions. You may find more detail here: `transcribe() <https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/r1.0.0rc1/asr/api.html#modules>`
```
Comment (titu1994) [inline docs/source/asr/results.rst:75]: No hard  coded paths to docs. Use relative paths with rst. Look at how I do relative path to Jasper block in Citrinet.

**Review by titu1994 — COMMENTED** [review]
For the asr part, minor change needed. 

Also a heads up - there are too many links to specific version of docs, it's gonna be difficult to update in the future.

File: docs/source/asr/results.rst (hunk 1 above)
Comment (VahidooX) [reply docs/source/asr/results.rst]: Done. Just need to test the docs.

File: docs/source/nlp/text_classification.rst [hunk 2]
```diff
@@ -1,12 +1,169 @@
...
+Text Classification Model
+=========================
 
-We recommend you try this model in a Jupyter notebook \
-(can run on `Google's Colab <https://colab.research.google.com/notebooks/intro.ipynb>`_.): \
-`NeMo/tutorials/nlp/Text_Classification_Sentiment_Analysis.ipynb <https://github.com/NVIDIA/NeMo/blob/main/tutorials/nlp/Text_Classification_Sentiment_Analysis.ipynb>`__.
+Text Classification Model is a sequence classification model based on BERT-based encoders. It can be used for a
+variety of tasks like text classification, sentiment analysis, domain/intent detection for dialogue systems, etc.
+The model takes a text input and classifies it into predefined categories. Most of the BERT-based encoders
+supported by HuggingFace including BERT, Megatron-LM, RoBERTa, DistilBERT, XLNet, etc can be used as the encoder of this model.
```
[the remark this answers is not in the document]
Comment (ekmb) [reply docs/source/nlp/text_classification.rst:9]: How about mentioning that multi-class but not multi-label tasks are supported?

File: docs/source/nlp/text_classification.rst [hunk 3]
```diff
@@ -1,12 +1,169 @@
...
+    lend some dignity to a dumb story   0
+    the greatest musicians  1
+
+You may need separate files for train, validation and test with this format.
+
+Dataset Conversion
+------------------
+
+If your dataset is stored in another format, you need to convert it to NeMo's format to use this model.
+There are some conversion scripts available for datasets: SST2, IMDB, ChemProt, and THUCnews. They can to convert them from their original format to NeMo's format.
```
Comment (ekmb) [inline docs/source/nlp/text_classification.rst:42]: links to the datasets?

File: docs/source/nlp/text_classification.rst [hunk 4]
```diff
@@ -1,12 +1,169 @@
...
+The config file contains three main sections:
+- trainer: Trainer section contains the configs for PTL training and you may find more info at
+:doc:`../../introduction/core.html#model-training` and
+`PTL Trainer class API <https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#trainer-class-api>'.
+- exp_manager: the configs of experiment manager. You can find more info at :doc:`../../introduction/core.html#experiment-manager`
+- model: contains the configs of the datasets, model architecture, tokenizer, optimizer, scheduler, etc.
+
+Some sub-sections of the model section are shared among most of the NLP models. The details of these sections can be found at:
+- tokenizer and language_model: The configs of the tokenizer and encoder part and the :doc:`../nlp_model.html#important-parameters`
+- optim: the configs of the optimizer and scheduler :doc:`../../introduction/core.html`
```
Comment (ekmb) [inline docs/source/nlp/text_classification.rst:85]: Really nice how you added also these links, do you mind adding this section to the NLPModel.rst, so the other models can reuse it?

File: docs/source/nlp/text_classification.rst [hunk 5]
```diff
@@ -1,12 +1,169 @@
...
+
+
+By default, the final model after training is saved in the path specified by 'NEMO_FILE_PATH'.
+
+Model Arguments
+^^^^^^^^^^^^^^^
+The following table lists some of the model's parameters you may use in the config files and set them from command line when training a model:
+
++-------------------------------------------+-----------------+------------------------------------------------+--------------------------------------------------------------------------------------------------------------+
+| **Parameter**                             | **Data Type**   |   **Default**                                  | **Description**                                                                                              |
```
Comment (ekmb) [inline docs/source/nlp/text_classification.rst:108]: you might want to drop the defaults as all these values are in the referenced config

**Review by ekmb — CHANGES_REQUESTED** [review]

File: docs/source/nlp/text_classification.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (VahidooX) [reply docs/source/nlp/text_classification.rst:9]: Wrote it in a way to convey that it classifies the whole sequence, not each token.

File: docs/source/nlp/text_classification.rst (hunk 5 above)
Comment (VahidooX) [reply docs/source/nlp/text_classification.rst:108]: Actually these defaults should be different from the config file. They should be the values if not set by the config file. In most cases, they are the same, but not always. Currently these values are the defaults of the config file. How about putting the defaults if they are not set in the config?

File: docs/source/nlp/text_classification.rst (hunk 4 above)
Comment (VahidooX) [reply docs/source/nlp/text_classification.rst:85]: You mean the links to the optim, tokenizer, and language model?

Comment (ekmb) [reply docs/source/nlp/text_classification.rst:85]: I meant links to the core/trainer sections. Some other models are referring to NlpModel.rst, would be nice to have them all in one place and not copy to all the models.rst files.

File: docs/source/asr/results.rst [hunk 6]
```diff
@@ -63,6 +63,19 @@ If you would like to programatically list the models available for a particular
   nemo_asr.models.<MODEL_BASE_CLASS>.list_available_models()
 
 
+Transcribing/Inference
```
Comment (fayejf) [inline docs/source/asr/results.rst:66]: Could we move Transcribing/Inference to the upper level?

Comment (fayejf) [reply docs/source/asr/results.rst:66]: nvm

File: docs/source/nlp/text_classification.rst (hunk 3 above)
Comment (VahidooX) [reply docs/source/nlp/text_classification.rst]: done.

**Review by ekmb — APPROVED** [review]

**Pull request merged**